validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGraffiti.java

```diff
@@ -0,0 +1,121 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.validator.client.KeyManager;
+import tech.pegasys.teku.validator.client.Validator;
+
+public class GetGraffiti extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v1/validator/{pubkey}/graffiti";
+  private final KeyManager keyManager;
+
+  private static final SerializableTypeDefinition<GraffitiResponse> GRAFFITI_TYPE =
+      SerializableTypeDefinition.object(GraffitiResponse.class)
+          .withOptionalField("pubkey", STRING_TYPE, GraffitiResponse::getPublicKey)
+          .withField("graffiti", STRING_TYPE, GraffitiResponse::getGraffiti)
+          .build();
+
+  private static final SerializableTypeDefinition<GraffitiResponse> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(GraffitiResponse.class)
+          .name("GraffitiResponse")
+          .withField("data", GRAFFITI_TYPE, Function.identity())
+          .build();
+
+  public GetGraffiti(final KeyManager keyManager) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getGraffiti")
+            .summary("Get Graffiti")
+            .description(
+                "Get the graffiti for an individual validator. If no graffiti is set explicitly, returns the process-wide default.")
+            .tags(TAG_GRAFFITI)
+            .withBearerAuthSecurity()
+            .pathParam(PARAM_PUBKEY_TYPE)
+            .response(SC_OK, "Success response", RESPONSE_TYPE)
+            .withAuthenticationResponses()
+            .withNotFoundResponse()
+            .build());
+    this.keyManager = keyManager;
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final BLSPublicKey publicKey = request.getPathParameter(PARAM_PUBKEY_TYPE);
+
+    final Optional<Validator> maybeValidator = keyManager.getValidatorByPublicKey(publicKey);
+    if (maybeValidator.isEmpty()) {
+      request.respondError(SC_NOT_FOUND, "Validator not found");
+      return;
+    }
+
+    String graffiti = maybeValidator.get().getGraffiti().map(this::processGraffitiBytes).orElse("");
+    request.respondOk(new GraffitiResponse(publicKey, graffiti));
+  }
+
+  private String processGraffitiBytes(final Bytes32 graffiti) {
+    return new String(graffiti.toArrayUnsafe(), StandardCharsets.UTF_8).strip().replace("\0", "");
+  }
+
+  static class GraffitiResponse {
+    private final Optional<String> publicKey;
```

Comment:
generally speaking we'd be having Optional<Bytes32> here i think in most objects, not sure why we're going the string route. String at rendering is fine but we lose information by making this string here.